src/components/ProductList.vue

```diff
@@ -45,12 +46,18 @@ export default {
       ],
       showForm: false,
       productIndex: '',
-      showDeleteDialog: false
+      showDeleteDialog: false,
+      showMessage: false
```

Comment:
Move this to a computed property and assign the boolean value from there. In the computed property you are checking the product array length, So that every time when it restarts it will check for array length and show message accordingly.
Since we declare the things inside a computed property, the message will appear and disappear accordingly with the state change.